### ISSUE TITLE
Add a rule to validate unresolved AsyncAPI document

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2261]
**Chore summary**

As a user I want to know if I defined `$ref` fields in proper places in my AsyncAPI document

**Is your feature request related to a problem?**

The following AsyncAPI document is invalid and Spectral should throw errors due to invalid defined references (in impossible places):

```yaml
asyncapi: '2.0.0'
channels: 
  someChannel:
    publish:
      $ref: '#/components/x-operations/someOperation'
components:
  'x-operations':
    someOperation: ...
```

We cannot use `$ref` in operations.

**Additional context**
Part of #2100
